weather_dl/download_pipeline/parsers.py

```diff
@@ -341,3 +345,20 @@ def require(condition: bool, message: str) -> None:
     config['parameters']['partition_keys'] = partition_keys
 
     return config
+
+
+def prepare_target_name(config: t.Dict) -> str:
```

Comment:
Yes, it can be simplified -- In fact, there are plans to get rid of the feature that this supports in an outstanding issue (will link to later).

In any case, I'll work on refactoring this now.

Reply:
Made an attempt to improve this, PTAL.